File: java/placedOrdersStatistics/src/validator/Validator.java
package validator;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class Validator {
    public static boolean ValidateParameters(String[] args) {
        if(args.length != 2) {
            System.out.println("you need to enter two dates:\n" +
                    "Start Date and End Date");
            return true;
        }
        return false;
    }

    public static boolean incorrectDates(String[] args) {
        try {
            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
            LocalDate.parse(args[0], formatter);
            LocalDate.parse(args[1], formatter);
            return false;
        } catch (Exception e) {
            System.out.println("An error occurred when trying to use the arguments passed.\nit is necessary to pass using the format: \"YYYY-MM-DD\"");
        }
        return true;
    }

    public static boolean EndIsBiggerThanStart(int start, int end) {
        if (end > start) {
            return true;
        } else {
            throw new IllegalArgumentException("The value of 'end' must be greater than 'start'.");
        }
    }




}

File: scala/src/main/scala/validator/Validator.scala
package validator

import java.time.LocalDate
import java.time.format.{DateTimeFormatter, DateTimeParseException}

object Validator {
  def validateParameters(args: Array[String])= {
    if(args.length != 2) {
      throw new IllegalArgumentException("you need to enter two dates:\n" +
        "Start Date and End Date");
    }
    false

  }

  def incorrectDates(args: Array[String]): Boolean = {
    try {
      val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
      val start = LocalDate.parse(args(0), formatter)
      val end = LocalDate.parse(args(1), formatter)

      if (start.isAfter(end)) {
        throw new IllegalArgumentException("The start date cannot be later than the end date.")
        return true
      }

      false
    } catch {
      case e: DateTimeParseException => {
        println("An error occurred when trying to use the arguments passed.\nit is necessary to use the format: \"YYYY-MM-DD\"")
        true
      }
    }
  }

  def EndIsBiggerThanStart(start: Int, end: Int): Boolean = {
    if(end > start) {
      true
    } else {
      throw new IllegalArgumentException("The value of 'end' must be greater than 'start'.");
    }
  }


}
